File: TAPKitReactNative/TAPKitModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TAPKitModule, NSObject)

RCT_EXTERN_METHOD(startScanning:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(stopScanning:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(connect:(NSString *)deviceId
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(disconnect:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(sendCommand:(NSDictionary *)command
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end 

File: TAPKitReactNative/TAPKitModule.swift
import Foundation
import React

@objc(TAPKitModule)
class TAPKitModule: NSObject {
  private var central: TAPKitCentral!
  
  override init() {
    super.init()
    central = TAPKitCentral()
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc(startScanning:rejecter:)
  func startScanning(_ resolve: @escaping RCTPromiseResolveBlock,
                    rejecter reject: @escaping RCTPromiseRejectBlock) {
    central.startScanning()
    resolve(nil)
  }
  
  @objc(stopScanning:rejecter:)
  func stopScanning(_ resolve: @escaping RCTPromiseResolveBlock,
                   rejecter reject: @escaping RCTPromiseRejectBlock) {
    central.stopScanning()
    resolve(nil)
  }
  
  @objc(connect:resolver:rejecter:)
  func connect(_ deviceId: String,
              resolver resolve: @escaping RCTPromiseResolveBlock,
              rejecter reject: @escaping RCTPromiseRejectBlock) {
    // Implementation for connecting to a specific device
    resolve(nil)
  }
  
  @objc(disconnect:rejecter:)
  func disconnect(_ resolve: @escaping RCTPromiseResolveBlock,
                 rejecter reject: @escaping RCTPromiseRejectBlock) {
    // Implementation for disconnecting
    resolve(nil)
  }
  
  @objc(sendCommand:resolver:rejecter:)
  func sendCommand(_ command: [String: Any],
                  resolver resolve: @escaping RCTPromiseResolveBlock,
                  rejecter reject: @escaping RCTPromiseRejectBlock) {
    // Implementation for sending commands to the device
    resolve(nil)
  }
} 